informalsystems/hermes
# Add missing CLIs for channel upgrade

Closes: #XXX

## Description




This PR introduces 4 `tx` CLIs for each step of a channel upgrade:

* `hermes tx chan-upgrade-init`
* `hermes tx chan-upgrade-try`
* `hermes tx chan-upgrade-ack`
* `hermes tx chan-upgrade-open`

## Testing

In order to test the CLIs use the following instructions:

NOTE: All CLIs should be called from the `simd-scripts` root

* Install locally simapp from the branch `04-channel-upgrades`
* Clone the forked `simd-scripts` https://github.com/ljoss17/simd-scripts and checkout the branch `luca_joss/channel-upgrade-setup`
* Call `make init`
* Create a channel with `hermes --config network/hermes/config.toml create channel --a-chain ibc1 --b-chain ibc2 --a-port transfer --b-port transfer --new-client-connection --yes`
* (Optional) Confirm the channel is created and the version is `ics20-1` with `hermes --config network/hermes/config.toml query channels --chain ibc1 --verbose`
* Init step for channel upgrade with: `hermes --config network/hermes/config.toml tx chan-upgrade-init --dst-chain ibc1 --dst-channel channel-0 --dst-port transfer --src-chain ibc1 --version "{\"fee_version\":\"ics29-1\",\"app_version\":\"ics20-1\"}" --timeout-time 200000`
* (Optional) Confirm the channel is in `InitUpgrade` state with `hermes --config network/hermes/config.toml query channels --chain ibc1 --verbose`
* Try step for channel upgrade with: `hermes --config network/hermes/config.toml tx chan-upgrade-try --dst-chain ibc2 --dst-connection connection-1 --dst-port transfer --src-chain ibc1 --src-channel channel-0 --src-port transfer --timeout-time 200000 --dst-channel channel-0`
* (Optional) Confirm the counterparty channel is in `TryUpgrade` state with `hermes --config network/hermes/config.toml query channels --chain ibc1 --verbose`
* Ack step for channel upgrade with: `hermes --config network/hermes/config.toml tx chan-upgrade-ack --dst-chain ibc1 --dst-connection connection-1 --dst-port transfer --src-chain ibc2 --src-channel channel-0 --src-port transfer --timeout-time 200000 --dst-channel channel-0`
* (Optional) Confirm the channel is in `Open` state and the version is now ` Version("{\"fee_version\":\"ics29-1\",\"app_version\":\"ics20-1\"}")` with `hermes --config network/hermes/config.toml query channels --chain ibc1 --verbose`
* Open step for counterparty channel end with: `hermes --config network/hermes/config.toml tx chan-upgrade-open --dst-chain ibc2 --dst-connection connection-1 --dst-port transfer --src-chain ibc1 --src-channel channel-0 --src-port transfer --timeout-time 200000 --dst-channel channel-0`
* (Optional) Confirm the counterparty channel end is in `Open` state and the version is now ` Version("{\"fee_version\":\"ics29-1\",\"app_version\":\"ics20-1\"}")` with `hermes --config network/hermes/config.toml query channels --chain ibc1 --verbose`

______

### PR author checklist:
- [ ] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:
